### PR TITLE
Optional param on waffle_flag and waffle_switch decorators to redirect to a view

### PR DESCRIPTION
**Pull request by rodgomes**
Introducing an optional param on waffle_flag and waffle_switch decorators to redirect to a view when the toggle is inactive, instead of throw a 404 page.

This way you can write a view like:

``` python
@waffle_switch('switch_name', 'url_name_to_redirect_to')
def myview(request):
    pass
```

Or 

``` python
@waffle_switch('switch_name', view_function)
def myview(request):
    pass
```

The same applies to flags.

Only affects mentioned decorators. Relevant test and documentation were updated as well

### CODE REVIEW

File: docs/usage/decorators.rst [hunk 1]
```diff
@@ -10,7 +10,7 @@ always-random nature, no decorator is provided for :ref:`samples
 <types-sample>`.)
 
 When the flag or switch is active, the view executes normally. When it
-is inactive, the view returns a 404.
+is inactive, the view returns a 404. Optionally, you can provide a view or URL name where the decorator can redirect to if you don't want to show a 404 page when the flag or switch is inactive.
```
Comment (jsocol) [inline docs/usage/decorators.rst]: mind line-wrapping this to 72 (or 80, I don't remember exactly) columns, like the rest of the file?

**Comment by jsocol** [comment]
Hey @rodgomes sorry for the delay on this and the issue. I do think this is a good addition. I just have the one minor comment in the docs.

**Comment by rodgomes** [comment]
Hi @jsocol, 
Sorry about that, my bad. I've fixed the docs line-wrapping.

**Comment by jsocol** [comment]
Thanks, @rodgomes!

**Pull request merged**